modules/workspace.nix

```diff
@@ -31,18 +31,29 @@ let
     type = 6;
   };
 
+  mouseActions = {
+    applicationLauncher = "org.kde.applauncher";
+    contextMenu = "org.kde.contextmenu";
+    paste = "org.kde.paste";
+    switchActivity = "switchactivity";
+    switchVirtualDesktop = "org.kde.switchdesktop";
+    switchWindow = "switchwindow";
+  };
+
+  mouseActionNames = lib.types.enum (builtins.attrNames mouseActions);
+
   anyThemeSet = (cfg.workspace.theme != null ||
     cfg.workspace.colorScheme != null ||
     (cfg.workspace.cursor != null && cfg.workspace.cursor.theme != null) ||
     cfg.workspace.lookAndFeel != null ||
     cfg.workspace.iconTheme != null);
 
-  # Becomes true if any option under "cfg.workspace.desktop.icons" is set to something other than null.
+  # Becomes true if any option under "cfg.workspace.desktop" is set to something other than null.
   anyDesktopFolderSettingsSet =
     let
       recurse = l: lib.any (v: if builtins.isAttrs v then recurse v else v != null) (builtins.attrValues l);
     in
-    recurse cfg.workspace.desktop.icons;
+    recurse cfg.workspace.desktop;
```

Comment:
Since it now includes mouseactions I think this should be renamed. `anyDesktopSettingsSet` could maybe work (or something more fitting if found).

Reply:
The name is based on the settings window that contains the settings for the desktop icons and the mouse actions, so I think it is still _correct_ in a way, but I also think `Desktop Settings` would be a better name for this window, so I’m not at all against this change. It breaks with Plasma terminology, but other parts of this PR do to (context menu vs. standard menu).

![20240811_173038](https://github.com/user-attachments/assets/6396fc44-928f-432a-8299-a4134a12bfc3)